scenes/game/game.gd

```diff
@@ -3046,6 +3100,85 @@ func _update_buttons():
 			instructions_wild_swing_allowed = false
 			button_choices.append({ "text": strike_text, "action": _on_shortcut_strike_pressed, "disabled": not can_strike or not game_wrapper.can_do_strike(Enums.PlayerId.PlayerId_Player) })
 			button_choices.append({ "text": boost_text, "action": _on_shortcut_boost_pressed, "disabled": not can_boost or not game_wrapper.can_do_boost(Enums.PlayerId.PlayerId_Player) })
+
+			# Check for character actions with card-related shortcuts
+			for i in range(game_wrapper.get_player_character_action_count(Enums.PlayerId.PlayerId_Player)):
+				var char_action = game_wrapper.get_player_character_action(Enums.PlayerId.PlayerId_Player, i)
+				if 'shortcut_effect_type' in char_action:
+					if char_action['shortcut_effect_type'] == "strike":
+						var action_possible = game_wrapper.can_do_character_action(Enums.PlayerId.PlayerId_Player, i)
+						var action_name = "Strike with Character Action"
```

Comment:
feels like there is a ton of duplicated code between each of these ifs

Reply:
Can you just have a table to map shortcut effect to action name and string or something?
You'll still have to set up like mincards and stuff in the ifs, but you can move a lot out that way.

Reply:
things that can be outside the big if block:
action possible
get action name from a table or char_action['action_name'] or default to 'Character Action'
reading force_cost/gauge_cost

Reply:
Didn't quite set up a table because of the conditions, but let me know if it's a bit cleaner now